packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +199,132 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The role for this CodeDeploy to use when deploying to this deployment group.
+   */
+  public readonly role: iam.IRole;
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
```

Comment:
Should be AWSCodeDeployRoleForECSLimited

Reply:
TIL - fixed.